File: Dataset/Leetcode/valid/66/642.cpp
class Solution {
public:
    vector<int> XXX(vector<int>& digits) {
        reverse(digits.begin(), digits.end());
        int add = 1;
        for (int i = 0; i < digits.size(); ++i)
        {
            digits[i] += add;
            if (digits[i] >= 10) add = 1, digits[i] -= 10;
            else add = 0;
        }
        if (add) digits.push_back(add);
        reverse(digits.begin(), digits.end());
        return digits;
    }
};

File: Dataset/Leetcode/train/78/642.java
 class Solution {
    public List<List<Integer>> XXX(int[] nums) {
        int len = nums.length;
        List<List<Integer>> ans = new ArrayList<>();
        int Max = (int)Math.pow(2,len);
        for(int i=0;i<Max;i++){
            int pre = 0;
            int number = i;
            List<Integer> temp = new ArrayList<>();
            while(number != 0){
                if((number&1) == 1){
                    temp.add(nums[pre]);
                }
                number>>=1;
                pre++;
            }
            ans.add(temp);
        }
        return ans;
    }
}

